src/lib/Scenes/Onboarding/OnboardingLogin.tests.tsx

```diff
@@ -123,4 +127,19 @@ describe("OnboardingLogin", () => {
       expect(passwordInput.props.autoFocus).toBe(true)
     })
   })
+
+  describe("Logic", () => {
+    it("Logs in and requests for push notifications permission when it's not yet determined", () => {
+      mockFetchNotificationPermissions(false).mockImplementationOnce((cb) =>
+        cb(null, PushAuthorizationStatus.NotDetermined)
+      )
+
+      const { getByTestId } = renderWithWrappersTL(<TestProvider />)
+
+      const loginButton = getByTestId("loginButton")
+      fireEvent.press(loginButton)
+
+      expect(LegacyNativeModules.ARTemporaryAPIModule.requestPrepromptNotificationPermissions).toBeCalled()
+    })
```

Comment:
Tried it with jest timers but no luck first, then second approach was `await waitFor(() => expectation)` but weirdly this didn't work.

What about testing it like this since every time it's gonna be 3 secs? (tests pass like this) @pvinis 
```suggestion
       setTimeout(() => {
         expect(LegacyNativeModules.ARTemporaryAPIModule.requestPrepromptNotificationPermissions).toBeCalled()
      }, 3000)
```

Reply:
@gkartalis This is not exactly doing any assertions. The setTimeout callback here will never be called, so the test will always pass.

Reply:
😬 indeed 😬